bin/dl-serve

```diff
@@ -1,60 +1,96 @@
-    the_config = yaml.safe_load(open("../examples/kv_store_tutorial.yaml", "r"))
-
-    the_endpoints = the_config.pop("endpoints", [])
-
-    the_service = create_object( the_config )
-    the_service.enable_scheduling = True
-    sig_handler.add_cancelable(the_service)
-
-    all_endpoints = []
-    for an_endpoint_conf in the_endpoints:
-        an_endpoint = create_object(an_endpoint_conf)
-        the_service.add_child( an_endpoint )
-        all_endpoints.append(an_endpoint)
-        if getattr(an_endpoint, 'log_interval', 0) > datetime.timedelta(seconds=0):
-            print("queue up start logging for '{}'".format(an_endpoint.name))
-            #action_id = the_service.schedule(an_endpoint.start_logging, datetime.datetime.now())
-            an_endpoint.start_logging()
-            #print("will start logging at {}".format(datetime.datetime.now() + the_service.execution_buffer*3))
-            #action_id = the_service.schedule(an_endpoint.scheduled_log, an_endpoint.log_interval, datetime.datetime.now() + the_service.execution_buffer*3)
-
-    print("about to start the service")
-    the_service.start()
-    print("services started, now to listen")
-    the_service.listen()
+            self._run(the_app)
+        except:
+            print("got an error... need to implement handling")
+            print("traceback was:\n{}".format(traceback.format_exc()))
+            self.return_code = 100
+
+    def _run(self, the_app):
+        this_config = the_app.master_config
+        print('python got a master config:\n{}'.format(this_config))
+
+        sig_handler = scarab.SignalHandler()
+
+        #file_path = this_config.get_value('runtime-file', None)
+        #if file_path is None:
+        #    #TODO: shold be a better error, also can I make this a required flag?
+        #    raise ValueError("runtime file is required")
+        #run_config = yaml.safe_load(open(file_path, 'r'))
+        #print('found file & loaded')
+
+        run_config = this_config.to_python().get('runtime-config')
+        the_endpoints = run_config.pop("endpoints", [])
+
+        service_config = run_config.copy()
+        service_config.update({'config': this_config})
+
+        the_service = self.create_object( run_config )
+        the_service.enable_scheduling = True
+        sig_handler.add_cancelable(the_service)
+        print('service built')
+
+        all_endpoints = []
+        for an_endpoint_conf in the_endpoints:
+            an_endpoint = self.create_object(an_endpoint_conf)
+            the_service.add_child( an_endpoint )
+            all_endpoints.append(an_endpoint)
+            if getattr(an_endpoint, 'log_interval', 0) > datetime.timedelta(seconds=0):
+                print("queue up start logging for '{}'".format(an_endpoint.name))
+                an_endpoint.start_logging()
+
+        print("about to start the service")
+        the_service.start()
+        print("services started, now to listen")
+        the_service.listen()
```

Comment:
I think that `service::stop()` should be called after `listen()` to close things out cleanly.

Reply:
fix is the next line so github didn't recognize that it is done